zerver/tests/test_decorators.py

```diff
@@ -647,23 +647,36 @@ class TestLoggingErrorException(Exception):
 
         return mock.patch("logging.error", side_effect=TestLoggingErrorException)
 
-    def check_rate_limit_public_or_user_views(self, remote_addr: str, client_name: str) -> None:
-        META = {"REMOTE_ADDR": remote_addr}
-
-        request = HostRequestMock(client_name=client_name, meta_data=META)
+    def check_rate_limit_public_or_user_views(
+        self, remote_addr: str, client_name: str, expect_rate_limit: bool = False
```

Comment:
I think it'd be substantially better to make `expect_rate_limit` a mandatory kwarg; I think the function name is pretty confusing otherwise, since by default it checks that rate limiting is disabled.

Reply:
Fixed.